cmd/agent/app/reporter/grpc/builder.go

```diff
@@ -102,6 +102,9 @@ func (b *ConnBuilder) CreateConnection(logger *zap.Logger, mFactory metrics.Fact
 	dialOptions = append(dialOptions, grpc.WithUnaryInterceptor(grpc_retry.UnaryClientInterceptor(grpc_retry.WithMax(b.MaxRetry))))
 	dialOptions = append(dialOptions, b.AdditionalDialOptions...)
 
+	connCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
`CreateConnection` is not a blocking function, it will exit after setting things up and this defer will be executed. Is that what you intended?

Please move this specific change to a different PR, as other packages don't require substantial changes and the rest of this PR can be merged.

Reply:
>CreateConnection is not a blocking function, it will exit after setting things up and this defer will be executed. Is that what you intended?

ohh my mistake, i should have just returned the `cancel()` function

Reply:
>Please move this specific change to a different PR, as other packages don't require substantial changes and the rest of this PR can be merged.

Okay!